main/main.c

```diff
@@ -1,548 +1,193 @@
-        TansmitMessage.MessageID = 0x123;
-        TansmitMessage.MessageSEQ = messageSEQ;
-        TansmitMessage.ControlFlags = MCP251XFD_STANDARD_MESSAGE_ID;
-        TansmitMessage.DLC = 4;
-        TansmitMessage.PayloadData = &payloadData[0];
-
-        if(++messageSEQ > MCP2518FD_SEQUENCE_MAX) messageSEQ = 0;
 
-        printCANMessage(&TansmitMessage);
-
-        //Send message and flush
-        ErrorExt1 = MCP251XFD_TransmitMessageToFIFO(dev, &TansmitMessage, MCP251XFD_FIFO2, true);
-    }else{
-        ESP_LOGW("CANTX", "FIFO FULL!, Status: %d", FIFOstatus);
+    //Allocate array to store current task states
+    startArraySize = uxTaskGetNumberOfTasks() + ARRAY_SIZE_OFFSET;
+    startArray = malloc(sizeof(TaskStatus_t) * startArraySize);
+    if (startArray == NULL) {
+        ret = ESP_ERR_NO_MEM;
+        goto exit;
     }
-    
-    return ErrorExt1;
-}
-
-MCPError receiveMessage(MCP251XFD *dev, MCP251XFD_CANMessage *receivedMessage){
-    eERRORRESULT ErrorExt1 = ERR_OK;
-    eMCP251XFD_FIFOstatus FIFOstatus = 0;
-    ErrorExt1 = MCP251XFD_GetFIFOStatus(dev, MCP251XFD_FIFO1, &FIFOstatus); // First get FIFO1 status
-    if (ErrorExt1 != ERR_OK){
-        ESP_LOGE("CANRX","Error getting FIFO Status %d", ErrorExt1);
-        return ErrorExt1;
-    } 
-    if ((FIFOstatus & MCP251XFD_RX_FIFO_NOT_EMPTY) > 0) // Second check FIFO not empty
-    {
-        // uint8_t RxPayloadData[64]; // In this example, the FIFO1 have 64 bytes of payload
-        // MCP251XFD_CANMessage ReceivedMessage;
-        // receivedMessage.PayloadData = &RxPayloadData[0]; // Add receive payload data pointer to the message structure
-        // that will be received
-        ErrorExt1 = MCP251XFD_ReceiveMessageFromFIFO(dev, receivedMessage, MCP251XFD_PAYLOAD_8BYTE,
-        NULL, MCP251XFD_FIFO1);
-        if (ErrorExt1 == ERR_OK)
-        {   
-            return ERR_OK;
-            // printCANMessage(&receivedMessage);
-            // ESP_LOGI("CANRX", "%"PRIu32", Received Message #%"PRIu32" from 0x%08" PRIx32 " with %d bytes. ", MessageTimeStamp, ReceivedMessage.MessageSEQ, ReceivedMessage.MessageID, ReceivedMessage.DLC);
-        }
-    }else if(FIFOstatus != 0){
-        ESP_LOGI("CANRX","FIFO Status %d", FIFOstatus);
+    //Get current task states
+    startArraySize = uxTaskGetSystemState(startArray, startArraySize, &startRunTime);
+    if (startArraySize == 0) {
+        ret = ESP_ERR_INVALID_SIZE;
+        goto exit;
     }
 
-    return ERR__EMPTY_DATA;
-}
-
-uint32_t getNumItemsInQueue(RingbufHandle_t ringbuffer){
-    unsigned int count;
-    vRingbufferGetInfo(ringbuffer, NULL,NULL,NULL,NULL, &count);
-    // printf("free: %u, read: %u, write: %u, acquire: %u, count, %u\n", freebytes, readbytes, writebytes, acquirebytes, count);
-    return count;
-}
-
-MCPError handleMessageReceive(MCP251XFD *dev, RingbufHandle_t ringbuffer){
-    eERRORRESULT ErrorExt1 = ERR_OK;
-    eMCP251XFD_FIFOstatus FIFOstatus = 0;
+    vTaskDelay(xTicksToWait);
```

Comment:
How long does this wait? Is this wait just used to measure statistics?

Reply:
As long a you want. In the main I have it set to 1 second. This is just to measure task times on the cpu to give a crude htop of the system. It is only called when the code is compile a a DEBUG log level.

Reply:
That seems fine